### PR TITLE
[release] minimalistic changes for 8.18

### PR DESCRIPTION
**Pull request by gares**
I got to write this by personal communication with @mattam82 

I trimmed some info which in my opinion don't belong to the changelog.

### CODE REVIEW

**Comment by proux01** [comment]
It seems this could highlight:
* work on Ltac2 (but I'm no expert and don't know what to put forward)
* improvements on the warning system, enabling finer control of warnings, particularly for deprecations
* some work on the stdlib (but again, I don't know what to put forward)

**Comment by gares** [comment]
I don't know exactly how Matthieu was doing his selection of items.

IMO we should put forward items which may have a big impact (on porting the code, or writing new code).  This is what I'd like to see in the changelog foreword. So I selected a change of default (long awaited, announced in advance, etc...), but maybe there are other change any user should be aware of.

There is also another document some projects produce at release time, which is somewhat mixed with the changelog in Coq. Examples that come to mind are GNOME, FireFox and VSCode which give a page with a few highlights, typically with a short video or screenshot to show "cool stuff". They really do a good job, but I'd keep advertisements out of the changelog.

**Review by Zimmi48 — DISMISSED** [review]
We've merged the release highlights and the changelog indeed. Maybe this was a bad idea. I personally find it easier to browse the changelog and have the highlights at the beginning. However, what you propose here is a break from this tradition, and it doesn't allow figuring out the most important changes in the release or to give due credit to their authors.

**Comment by Zimmi48** [comment]
Note that the release summary (or highlights) is also what I typically copy over to Zenodo:

https://zenodo.org/record/8161141

**Comment by gares** [comment]
Then tell me what to write there, since I don't know.

**Comment by herbelin** [comment]
I have two remarks in passing:

- soundness fix -> soundness fixes

  Maybe also tell in the summary in which part of the kernel are the soundness bugs.

- change Miscellaneous into Extraction, since there is only one item there, which is about extraction

  By the way, regarding changes in extraction, I'm regularly hesitating about which category to use (typically between vernac-commands-and-options and misc). What about having an explicit Extraction category?

**Comment by herbelin** [comment]
By the way, still in passing, the quality of the release job is impressing.

**Comment by Zimmi48** [comment]
> * What about having an explicit Extraction category?

Sure, that's a good idea. We can remove the category from the changelog when it is empty anyway.

**Comment by herbelin** [comment]
> Sure, that's a good idea.

Do you want me to do it?

**Comment by Zimmi48** [comment]
Yes, please!

**Comment by herbelin** [comment]
Maybe an excès de zèle, but since I had the opportunity to look at the change log, I seized it (the opportunity) to apply the `Describe your change here but do not end with a period` rule of `dev/tools/make-changelog.sh` (even retroactively, if nobody objects to this syntactic instance of "historical revisionism"). This is PR #18001.

**Comment by gares** [comment]
> Then tell me what to write there, since I don't know.

Dear @coq/doc-maintainers here my attempt at identifying some highlights. Quick feedback is welcome.

File: doc/sphinx/changes.rst
Comment (proux01) [inline doc/sphinx/changes.rst:18]:
```suggestion
with a host of improvements. We highlight a few impactful changes above all:
```

File: doc/sphinx/changes.rst
Comment (Zimmi48) [inline doc/sphinx/changes.rst:23]:
```suggestion
    sort (the algorithm used to pick `Type`). Thanks to this feature many `Prop`
```

File: doc/sphinx/changes.rst
Comment (Zimmi48) [inline doc/sphinx/changes.rst:26]:
```suggestion
    as names (of contants, inductive types, etc) and finally fine-grained
```

File: doc/sphinx/changes.rst
Comment (herbelin) [inline doc/sphinx/changes.rst:29]:
```suggestion
    This should be particularly useful to handle deprecations.
```

**Review by Zimmi48 — COMMENTED** [review]
Thanks! And sorry for not helping more.

**Comment by Zimmi48** [comment]
Unless we discuss it in a Coq Call (at the very least), I would keep the other parts of the release summary, however. In particular, we do not currently have another place to give credits (that would be kept up-to-date). The permalinks to this particular version documentation are useful and the number of merged PRs and closed issues in the milestone is also interesting information (even if very approximate).

File: doc/sphinx/changes.rst
Comment (gares) [reply doc/sphinx/changes.rst]: Apparently it is broken (see https://github.com/coq/coq/issues/17982) so I'm not putting it forward.

Comment (Zimmi48) [reply doc/sphinx/changes.rst]: OK that makes sense, although it works for quite a few useful things.

File: doc/sphinx/changes.rst
Comment (herbelin) [inline doc/sphinx/changes.rst:26]:
```suggestion
    as names (of constants, inductive types, etc) and finally fine-grained
```

**Comment by Zimmi48** [comment]
For tasks other than committing code (to the Coq repository) basically.

>All PR are listed with their authors

Note that even this is not true since many PRs are silent ones (from a changelog perspective) because they do not contain user-facing changes, although it is true that the committers will still be listed in the list that you've put in the summary.

**Comment by gares** [comment]
> For tasks other than committing code (to the Coq repository) basically.

I'm not competent for doing this.
I did not follow everything that happened in the Coq community and I'm not qualified to pick the most valuable contributions. After all I'm managing the release, not the community nor coordinating the project.

So you can either live with a technical foreword written and signed by me, or find someone coauthoring it (quickly).

The fact that it does not take 1h to complete this task and that this task is in the critical path for the release is, to me, a sign that something is off with it. We organized the release so that this task has weeks to complete, but if it does not for any good or bad reason, we should just live with it.

**Review by Zimmi48 — APPROVED** [review]
Will leave a bit of time before merging, so that others can finish reviewing / commenting.

**Comment by gares** [comment]
Nothing prevents you or Matthieu to update the doc after the tag.

**Comment by mattam82** [comment]
I think it would be better to let the release manager write the highlights in the future as well, as they are already familiar with all the changes and probably are aware of the main changes / compat issues. I’m also ok to remove the mentions on work on opam/platform/website etc in the changes, which are not really release-specific. But the stats on PRs and issues give a rough idea of the amount of changes and I think it would be good to keep them here.

**Comment by mattam82** [comment]
Also I would not become a blocking point here, and it feels fairer that I don’t take credit for releases I barely contributed to in practice while the release managment work is non-trivial.

File: doc/sphinx/changes.rst
Comment (gares) [inline doc/sphinx/changes.rst:43]:
```suggestion
The 41 contributors to the 8.18 version are:
Reynald Affeldt, Tanaka Akira, Matthieu Baty, Yves Bertot,
Lasse Blaauwbroek, Ana Borges, Kate Deplaix, Ali Caglayan, Cyril Cohen, Maxime Dénès,
Andrej Dudenhefner, Andres Erbsen, Jim Fehrle, Yannick Forster,
Paolo G. Giarrusso, Gaëtan Gilbert, Jason Gross, Samuel Gruetter,
Stefan Haan, Hugo Herbelin, Yoshihiro Imai, Emilio Jesús Gallego Arias,
Olivier Laurent, Meven Lennon-Bertrand, Rodolphe Lepigre, Yishuai Li,
Guillaume Melquiond, Karl Palmskog, Pierre-Marie Pédrot, Stefan Radziuk,
Ramkumar Ramachandra, Pierre Rousselin, Pierre Roux, Julin S,
Kazuhiko Sakaguchi, Weng Shiwei, Michael Soegtrop, Matthieu Sozeau,
Enrico Tassi, Hao Yang, Théo Zimmermann.
```

**Comment by Zimmi48** [comment]
Sounds good. We should probably also improve the tooling so that writing this summary is easier, including to highlight non-code specific tasks (e.g., starting with reviews).

File: doc/sphinx/changes.rst
Comment (gares) [inline doc/sphinx/changes.rst:46]:
```suggestion
We are very grateful to the Coq community at large for the help in making this
new version of Coq. Version 8.18's development spanned over 6 months from the release of
Coq 8.17.0. Maxime Dénès and Enrico Tassi were the release managers.
```

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:18]:
```suggestion
with a host of improvements. We highlight a few impactful changes:
```

Doesn't add anything that "highlighting" doesn't cover

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:22]:
```suggestion
  - the universe unification algorithm can now delay the commitment to a
```

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:23]:
```suggestion
    `Sort` (the algorithm used to pick `Type`). Thanks to this feature many `Prop`
```

IINM

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:26]:
```suggestion
    as names (of constants, inductive types, etc) and fine-grained
```

Not needed.

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:28]:
```suggestion
  - The warning system offers new categories, enabling finer (de)activation of specific warnings.
```

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:30]:
```suggestion
  - many new lemmas useful for teaching analysis with Coq are now part of
```

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:41]: @ju-sh, would you like us to include your full name in the 8.18 contributors list?

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:46]:
```suggestion
We are very grateful to the Coq community for their help in creating 8.18
in the 6 months since the release of
```

File: doc/sphinx/changes.rst
Comment (jfehrle) [inline doc/sphinx/changes.rst:49]:
```suggestion
| Sophia-Antipolis, September 2023,
```

**Review by jfehrle — COMMENTED** [review]
Wording suggestions

File: doc/sphinx/changes.rst
Comment (gares) [inline doc/sphinx/changes.rst:41]:
```suggestion
Ramkumar Ramachandra, Pierre Rousselin, Pierre Roux, Julin Shaji,
```

File: doc/sphinx/changes.rst
Comment (gares) [inline doc/sphinx/changes.rst:32]:
```suggestion
  - Many new lemmas useful for teaching analysis with Coq are now part of
    the standard library about real numbers.
  - The `#[deprecated]` attribute can now be applied to definitions.
  
```

**Comment by gares** [comment]
should be good to go

**Comment by gares** [comment]
please merge so that I can backport

**Comment by maximedenes** [comment]
@Zimmi48 merging because @gares is asking

**Comment by maximedenes** [comment]
@coqbot merge now

**Comment by maximedenes** [comment]
sudo @coqbot merge now

**Comment by maximedenes** [comment]
@coqbot Ok, fair enough, you won

**Comment by proux01** [comment]
@coqbot merge now

**Pull request merged**
